src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3020,10 +3020,17 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
             source = GalleryArtifactVersionSource(id=managed_image)
         if os_snapshot is not None:
             os_disk_image = GalleryOSDiskImage(source=GalleryArtifactVersionSource(id=os_snapshot))
+        if data_snapshot_luns and not data_snapshots:
+            raise CLIError('usage error: --data-snapshot-luns must be used together with --data-snapshots')
         if data_snapshots:
+            if data_snapshot_luns and len(data_snapshots) != len(data_snapshot_luns):
+                raise CLIError('usage error: Length of --data-snapshots and --data-snapshot-luns should be equal.')
```

Comment:
we validate the length of luns here, do we need add validation for the value of each lun ? For example, do we need to check whether there are duplicate luns(1, 1, 1) or invliad character(1, a, b) ? Or we just rely on the service to do the validation of these cases ?

Reply:
You are right. They should be non-negative integers.